src/main/java/com/powsybl/balances_adjustment/util/VoltageLevelsArea.java

```diff
@@ -8,30 +8,138 @@
 
 import com.powsybl.iidm.network.*;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
 /**
  * @author Ameni Walha {@literal <ameni.walha at rte-france.com>}
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  */
-public class VoltageLevelsArea extends AbstractNetworkArea {
-
-    private final String name;
+public class VoltageLevelsArea implements NetworkArea {
```

Comment:
I have exactly the same question about the cache management for this implementation